### PR TITLE
Don't return lease not found errors on platform mismatch

### PR DESCRIPTION
**Pull request by rumpl**
**- What I did**

#43939 changed the signature of `manifestMatchesPlatform` and started returning the lease error, the issue is that, if the image for the wanted platform doesn't exist, then the lease doesn't exist either. We shouldn't return an error if the lease is not found

**- How I did it**

By checking the error returned when we list lease resources.

**- How to verify it**

With this compose file

```yaml
services:
  test:
    build:
      context: .
      dockerfile: Dockerfile
```

And this Dockerfile:

```dockerfile
FROM alpine
```

Run the compose project once:

```console
$ docker compose up
```
Then run it but with a preferred platform specified:

Before:

```console
$ DOCKER_DEFAULT_PLATFORM=linux/arm64 docker-compose up
[+] Running 0/0
 ⠋ Container compose-platform-test-1  Recreate                                                                                                                         0.0s 
Error response from daemon: lease "moby-image-sha256:e22aab0781e952e8ae9c49b2162eda23ef9a62d26a7b1e6a2d66b77f6751c45e": not found
```
After:

```console
$ DOCKER_DEFAULT_PLATFORM=linux/arm64 docker-compose up
[+] Running 0/0
 ⠋ Container compose-platform-test-1  Recreate                                                                                                                         0.0s 
Error response from daemon: image with reference compose-platform-test was found but does not match the specified platform: wanted linux/arm64, actual: linux/amd64
```

This was reported in multiple places:

* https://github.com/laravel/sail/issues/597
* https://github.com/docker/compose/issues/10856#issuecomment-1682601800
* https://stackoverflow.com/questions/76413014/error-response-from-daemon-lease-moby-image-sha256c274-not-found

**- Description for the changelog**
Fixed the reported error when the local image's platform mismatches the wanted one

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/images/image.go [hunk 1]
```diff
@@ -51,13 +51,13 @@ func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, image str
...
 }
 
-func (i *ImageService) manifestMatchesPlatform(ctx context.Context, img *image.Image, platform ocispec.Platform) (bool, error) {
+func (i *ImageService) manifestMatchesPlatform(ctx context.Context, img *image.Image, platform ocispec.Platform) bool {
 	logger := log.G(ctx).WithField("image", img.ID).WithField("desiredPlatform", platforms.Format(platform))
 
 	ls, leaseErr := i.leases.ListResources(ctx, leases.Lease{ID: imageKey(img.ID().String())})
 	if leaseErr != nil {
 		logger.WithError(leaseErr).Error("Error looking up image leases")
-		return false, leaseErr
```
Comment (vvoland) [inline daemon/images/image.go:60]: Context cancellation will also result in the platform mismatch error.
How about leaving the function signature as is, and just change this line to return `nil` error?

Comment (vvoland) [reply daemon/images/image.go:60]: Also, we should handle context cancellation in the loop below (in a follow-up, as it's not strictly related to the subject of this PR).

Comment (rumpl) [reply daemon/images/image.go:60]: Okay changed it to return `false, nil` if the lease is not found and to return the error otherwise

**Review by vvoland — APPROVED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**